api/src/main/java/com/epam/esm/controller/TagController.java

```diff
@@ -0,0 +1,64 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.model.ErrorResponse;
+import com.epam.esm.model.dto.TagDto;
+import com.epam.esm.service.TagService;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/api/v1/tags")
+public class TagController {
+    private final TagService tagService;
+
+    @Autowired
+    public TagController(TagService tagService) {
+        this.tagService = tagService;
+    }
+
+    @GetMapping
+    @ApiOperation(value = "Api v1. Get all tags")
+    public ResponseEntity<List<TagDto>> allTags(@RequestParam(required = false) String name) {
+        List<TagDto> tags;
+
+        if (name != null){
+            tags = tagService.getTagByPartName(name);
+        }else {
+            tags = tagService.getAllTags();
+        }
+        return ResponseEntity.ok(tags);
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "Api v1. Get tag by id")
+    public ResponseEntity<TagDto> tagById(
+            @PathVariable Long id
+    ) {
+        TagDto tagById = tagService.getTagById(id);
+        return ResponseEntity.ok(tagById);
+    }
+
+    @PostMapping
+    @ApiOperation(value = "Api v1. Create tag")
+    public ResponseEntity<TagDto> create(
```

Comment:
in order to be consistent across the class if you have method `delete**Tag**`please rename this metho as `create**Tag**`